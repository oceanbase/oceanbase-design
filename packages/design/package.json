{
  "name": "@oceanbase/design",
  "version": "0.4.2",
  "description": "The Design System of OceanBase",
  "keywords": [
    "oceanbase",
    "design",
    "ui",
    "react",
    "component"
  ],
  "homepage": "https://github.com/oceanbase/oceanbase-design/packages/design",
  "repository": {
    "type": "git",
    "url": "git@github.com:oceanbase/oceanbase-design.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "main": "lib/index.js",
  "unpkg": "dist/design.min.js",
  "module": "es/index.js",
  "typings": "es/index.d.ts",
  "files": [
    "es",
    "lib",
    "dist",
    "package.json",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "build": "father build && cp src/style/reset.css dist/"
  },
  "dependencies": {
    "@ant-design/cssinjs": "^1.22.0",
    "@oceanbase/aliyun-theme": "^0.1.6",
    "@oceanbase/icons": "workspace:^",
    "@oceanbase/util": "workspace:^",
    "ahooks": "^2.10.14",
    "antd": "^5.22.1",
    "classnames": "^2.5.1",
    "lodash": "^4.17.21",
    "lottie-web": "^5.12.2",
    "prop-types": "^15.8.1",
    "rc-util": "^5.43.0",
    "react-sticky-mouse-tooltip": "^0.0.1"
  },
  "devDependencies": {
    "@babel/cli": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "antd-token-previewer": "^2.0.8",
    "rc-select": "^14.16.3",
    "rc-slider": "^11.1.7",
    "rc-table": "^7.48.1",
    "rc-tree-select": "^5.24.4"
  },
  "peerDependencies": {
    "react": ">=16.9.0",
    "react-dom": ">=16.9.0"
  }
}
