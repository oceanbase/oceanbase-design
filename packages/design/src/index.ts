import './global.css';

export * from 'antd';
export { version } from '../package.json';

export { default as Alert } from './alert';
export type { AlertProps } from './alert';

export { default as Button } from './button';
export type { ButtonProps } from './button';

export { default as Space } from './space';
export type { SpaceProps, SpaceSize } from './space';

export { Col, Row } from './grid';
export type { ColProps, RowProps, ColSize } from './grid';

export { default as Typography } from './typography';
export type { TypographyProps } from './typography';

export { default as Card } from './card';
export type { CardProps } from './card';

export { default as ConfigProvider } from './config-provider';
export type { ConfigProviderProps, ConfigConsumerProps, ThemeConfig } from './config-provider';

export { default as Descriptions } from './descriptions';
export type {
  DescriptionsProps,
  DescriptionsItemType,
  DescriptionsItemProps,
} from './descriptions';

export { default as Empty } from './empty';
export type { EmptyProps } from './empty';

export { default as Form } from './form';
export type { FormProps, FormItemProps } from './form';

export { default as Modal } from './modal';
export type { ModalProps, ModalFuncProps, ModalProgressProps } from './modal';

export { default as Drawer } from './drawer';
export type { DrawerProps } from './drawer';

export { message, notification, token } from './static-function';

export { default as List } from './list';
export type { ListProps } from './list';

export { default as Table } from './table';
export type { TableProps } from './table';

export { default as Tabs } from './tabs';
export type { TabsProps } from './tabs';

export { default as Tag } from './tag';
export type { TagProps } from './tag';

export { default as Input } from './input';
export type { InputProps } from './input';

export { default as InputNumber } from './input-number';
export type { InputNumberProps } from './input-number';

export { default as Radio } from './radio';
export type { RadioProps } from './radio';

export { default as Checkbox } from './checkbox';
export type { CheckboxProps } from './checkbox';

export { default as Switch } from './switch';
export type { SwitchProps } from './switch';

export { default as Select } from './select';
export type { SelectProps } from './select';

export { default as TreeSelect } from './tree-select';
export type { TreeSelectProps } from './tree-select';

export { default as Slider } from './slider';
export type { SliderSingleProps, SliderRangeProps } from './slider';

export { default as Tooltip } from './tooltip';
export type { TooltipProps } from './tooltip';

export { default as Segmented } from './segmented';
export type { SegmentedProps } from './segmented';

export { default as Breadcrumb } from './breadcrumb';
export type { BreadcrumbProps, BreadcrumbItemProps } from './breadcrumb';

export { default as Result } from './result';
export type { ResultProps, ResultType, ResultStatusType } from './result';

export { default as Spin } from './spin';
export type { SpinProps } from './spin';

export { default as Badge } from './badge';
export type { BadgeProps } from './badge';

export { default as Lottie } from './lottie';
export type { LottieProps, LottieRef } from './lottie';

export { default as theme } from './theme';

import theme from './theme';

const { useToken } = theme;
export { useToken };

export type { PresetStatusColorType } from 'antd/es/_util/colors';
export { genLargeStyle } from './_util/genStyle';
