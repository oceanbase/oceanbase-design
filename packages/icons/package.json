{
  "name": "@oceanbase/icons",
  "version": "0.4.1",
  "description": "The Icon Library for OceanBase",
  "keywords": [
    "oceanbase",
    "design",
    "icon",
    "react",
    "component"
  ],
  "homepage": "https://github.com/oceanbase/oceanbase-design/packages/icons",
  "repository": {
    "type": "git",
    "url": "git@github.com:oceanbase/oceanbase-design.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "main": "lib/index.js",
  "unpkg": "dist/design.min.js",
  "module": "es/index.js",
  "typings": "es/index.d.ts",
  "files": [
    "es",
    "lib",
    "dist",
    "package.json",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "build": "npm run generate && father build",
    "generate": "rm -rf src && ts-node scripts/generate.ts"
  },
  "dependencies": {
    "@ant-design/colors": "^7.1.0",
    "@ant-design/icons": "^5.5.1",
    "@oceanbase/icons-svg": "^1.0.10",
    "lodash.camelcase": "^4.3.0"
  },
  "peerDependencies": {
    "react": ">=16.9.0",
    "react-dom": ">=16.9.0"
  }
}
