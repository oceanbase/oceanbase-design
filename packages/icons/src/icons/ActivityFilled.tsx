// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import ActivityFilledSvg from '../asn/ActivityFilled';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const ActivityFilled = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={ActivityFilledSvg} />;

ActivityFilled.displayName = 'ActivityFilled';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(ActivityFilled);