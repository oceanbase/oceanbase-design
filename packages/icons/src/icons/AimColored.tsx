// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import AimColoredSvg from '../asn/AimColored';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const AimColored = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={AimColoredSvg} />;

AimColored.displayName = 'AimColored';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(AimColored);