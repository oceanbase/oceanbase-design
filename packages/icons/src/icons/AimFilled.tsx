// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import AimFilledSvg from '../asn/AimFilled';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const AimFilled = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={AimFilledSvg} />;

AimFilled.displayName = 'AimFilled';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(AimFilled);