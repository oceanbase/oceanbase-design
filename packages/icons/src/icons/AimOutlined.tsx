// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import AimOutlinedSvg from '../asn/AimOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const AimOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={AimOutlinedSvg} />;

AimOutlined.displayName = 'AimOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(AimOutlined);