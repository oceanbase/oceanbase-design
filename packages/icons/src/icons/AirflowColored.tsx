// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import AirflowColoredSvg from '../asn/AirflowColored';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const AirflowColored = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={AirflowColoredSvg} />;

AirflowColored.displayName = 'AirflowColored';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(AirflowColored);