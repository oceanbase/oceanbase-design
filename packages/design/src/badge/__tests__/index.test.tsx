import React from 'react';
import { render } from '@testing-library/react';
import { describe, it } from 'vitest';
import { Badge } from '@oceanbase/design';
import { TaobaoOutlined } from '@oceanbase/icons';

describe('Badge', () => {
  it('status dot', () => {
    const { container, asFragment } = render(<Badge status="success" text="Success" />);
    // badge
    expect(container.querySelector('.ant-badge')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status')).toBeTruthy();
    // status dot
    expect(container.querySelector('.ant-badge-status-dot')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status-success')).toBeTruthy();
    // status text
    expect(container.querySelector('.ant-badge-status-text')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status-text').textContent).toBe('Success');
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('status icon', () => {
    const { container, asFragment } = render(<Badge status="success" text="Success" icon={true} />);
    // badge
    expect(container.querySelector('.ant-badge')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status')).toBeTruthy();
    // status icon
    expect(container.querySelector('.ant-badge-status-icon')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status-success')).toBeTruthy();
    // status text
    expect(container.querySelector('.ant-badge-status-text')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status-text').textContent).toBe('Success');
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('status custom icon', () => {
    const { container, asFragment } = render(
      <Badge status="success" text="Success" icon={<TaobaoOutlined />} />
    );
    // badge
    expect(container.querySelector('.ant-badge')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status')).toBeTruthy();
    // status icon
    expect(container.querySelector('.ant-badge-status-icon')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status-success')).toBeTruthy();
    expect(container.querySelector('.anticon-taobao')).toBeTruthy();
    // status text
    expect(container.querySelector('.ant-badge-status-text')).toBeTruthy();
    expect(container.querySelector('.ant-badge-status-text').textContent).toBe('Success');
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
