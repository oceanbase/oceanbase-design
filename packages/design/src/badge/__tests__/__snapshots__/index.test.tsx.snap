// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > status custom icon 1`] = `
<span
  class="ant-badge ant-badge-status"
  style="display: inline-block;"
>
  <span
    class="ant-badge-status-icon ant-badge-status-success"
  >
    <span
      aria-label="taobao"
      class="anticon anticon-taobao"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="taobao"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M168.5 273.7a68.7 68.7 0 10137.4 0 68.7 68.7 0 10-137.4 0zm730 79.2s-23.7-184.4-426.9-70.1c17.3-30 25.6-49.5 25.6-49.5L396.4 205s-40.6 132.6-113 194.4c0 0 70.1 40.6 69.4 39.4 20.1-20.1 38.2-40.6 53.7-60.4 16.1-7 31.5-13.6 46.7-19.8-18.6 33.5-48.7 83.8-78.8 115.6l42.4 37s28.8-27.7 60.4-61.2h36v61.8H372.9v49.5h140.3v118.5c-1.7 0-3.6 0-5.4-.2-15.4-.7-39.5-3.3-49-18.2-11.5-18.1-3-51.5-2.4-71.9h-97l-3.4 1.8s-35.5 159.1 102.3 155.5c129.1 3.6 203-36 238.6-63.1l14.2 52.6 79.6-33.2-53.9-131.9-64.6 20.1 12.1 45.2c-16.6 12.4-35.6 21.7-56.2 28.4V561.3h137.1v-49.5H628.1V450h137.6v-49.5H521.3c17.6-21.4 31.5-41.1 35-53.6l-42.5-11.6c182.8-65.5 284.5-54.2 283.6 53.2v282.8s10.8 97.1-100.4 90.1l-60.2-12.9-14.2 57.1S882.5 880 903.7 680.2c21.3-200-5.2-327.3-5.2-327.3zm-707.4 18.3l-45.4 69.7 83.6 52.1s56 28.5 29.4 81.9C233.8 625.5 112 736.3 112 736.3l109 68.1c75.4-163.7 70.5-142 89.5-200.7 19.5-60.1 23.7-105.9-9.4-139.1-42.4-42.6-47-46.6-110-93.4z"
        />
      </svg>
    </span>
  </span>
  <span
    class="ant-badge-status-text"
  >
    Success
  </span>
</span>
`;

exports[`Badge > status dot 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-success"
  />
  <span
    class="ant-badge-status-text"
  >
    Success
  </span>
</span>
`;

exports[`Badge > status icon 1`] = `
<span
  class="ant-badge ant-badge-status"
  style="display: inline-block;"
>
  <span
    class="ant-badge-status-icon ant-badge-status-success"
  >
    <span
      aria-label="check-circle"
      class="anticon anticon-check-circle"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </span>
  </span>
  <span
    class="ant-badge-status-text"
  >
    Success
  </span>
</span>
`;
